### PR TITLE
CORE-19612 Publish session creation records from InboundMessageProcessor

### PR DESCRIPTION
**Pull request by YashNabar**
Changes the session event publisher to return records for sessions created to `InboundMessageProcessor` instead of publishing them itself.

Multi-cluster e2e test run: [link](https://ci02.dev.r3.com/job/Corda5/job/corda-e2e-tests-multi-cluster-tests/job/yash%252Ftest/)

### CODE REVIEW

File: components/link-manager/src/main/kotlin/net/corda/p2p/linkmanager/inbound/InboundMessageProcessor.kt [hunk 1]
```diff
@@ -128,27 +128,32 @@ internal class InboundMessageProcessor(
...
+                    )
+                }
+
+                null -> {
+                    TraceableItem(
+                        response.sessionCreationRecords, traceableMessage.originalRecord
+                    )
+                }
+
+                else -> {
```
Comment (williamvigorr3) [inline components/link-manager/src/main/kotlin/net/corda/p2p/linkmanager/inbound/InboundMessageProcessor.kt:154]: Can we combine this, with `ResponderHelloMessage`? They do the same thing.

File: components/link-manager/src/main/kotlin/net/corda/p2p/linkmanager/sessions/StatefulSessionManagerImpl.kt [hunk 2]
```diff
@@ -356,12 +357,12 @@ internal class StatefulSessionManagerImpl(
...
-                        sessionEventPublisher.sessionCreated(key, SessionDirection.OUTBOUND)
-                    }
+                        sessionEventPublisher.recordsForSessionCreated(key, SessionDirection.OUTBOUND)
+                    } ?: emptyList()
                 }
+                else -> emptyList()
             }
-        }.map { result ->
-            result.traceable to result.result?.message
+            result.traceable to SessionManager.ProcessSessionMessagesResult(result.result?.message, sessionCreationRecords)
```
Comment (williamvigorr3) [inline components/link-manager/src/main/kotlin/net/corda/p2p/linkmanager/sessions/StatefulSessionManagerImpl.kt]: `result.traceable to SessionManager.ProcessSessionMessagesResult(linkOutMessage, sessionCreationRecords)`

File: components/link-manager/src/main/kotlin/net/corda/p2p/linkmanager/sessions/events/StatefulSessionEventPublisher.kt [hunk 3]
```diff
@@ -26,8 +26,8 @@ class StatefulSessionEventPublisher(
     private val config = PublisherConfig(CLIENT_ID)
     private val publisher = PublisherWithDominoLogic(publisherFactory, coordinatorFactory, config, messagingConfiguration)
 
-    fun sessionCreated(key: String, direction: SessionDirection) {
-        publisher.publish(listOf(Record(Schemas.P2P.SESSION_EVENTS, key, SessionEvent(SessionCreated(direction, key)))))
+    fun recordsForSessionCreated(key: String, direction: SessionDirection): List<Record<String, *>> {
```
Comment (williamvigorr3) [inline components/link-manager/src/main/kotlin/net/corda/p2p/linkmanager/sessions/events/StatefulSessionEventPublisher.kt]: Nit: `List<Record<String, *>>` -> `List<Record<String, SessionEvent>>`?

File: components/link-manager/src/main/kotlin/net/corda/p2p/linkmanager/sessions/StatefulSessionManagerImpl.kt (hunk 2 above)
Comment (YashNabar) [reply components/link-manager/src/main/kotlin/net/corda/p2p/linkmanager/sessions/StatefulSessionManagerImpl.kt]: Applied in [7fb0d31](https://github.com/corda/corda-runtime-os/pull/5846/commits/7fb0d31c328c97d9ea306409966d9e4e8c1b4a59)

File: components/link-manager/src/main/kotlin/net/corda/p2p/linkmanager/sessions/events/StatefulSessionEventPublisher.kt (hunk 3 above)
Comment (YashNabar) [reply components/link-manager/src/main/kotlin/net/corda/p2p/linkmanager/sessions/events/StatefulSessionEventPublisher.kt]: Applied in [65b63af](https://github.com/corda/corda-runtime-os/pull/5846/commits/65b63af143e78ef40b18f6f3640a74ff6fa3d82d)

File: components/link-manager/src/main/kotlin/net/corda/p2p/linkmanager/inbound/InboundMessageProcessor.kt (hunk 1 above)
Comment (YashNabar) [reply components/link-manager/src/main/kotlin/net/corda/p2p/linkmanager/inbound/InboundMessageProcessor.kt:154]: Changed in [65b63af](https://github.com/corda/corda-runtime-os/pull/5846/commits/65b63af143e78ef40b18f6f3640a74ff6fa3d82d)

File: components/link-manager/src/main/kotlin/net/corda/p2p/linkmanager/sessions/events/StatefulSessionEventPublisher.kt [hunk 4]
```diff
@@ -26,8 +26,8 @@ class StatefulSessionEventPublisher(
     private val config = PublisherConfig(CLIENT_ID)
     private val publisher = PublisherWithDominoLogic(publisherFactory, coordinatorFactory, config, messagingConfiguration)
 
-    fun sessionCreated(key: String, direction: SessionDirection) {
-        publisher.publish(listOf(Record(Schemas.P2P.SESSION_EVENTS, key, SessionEvent(SessionCreated(direction, key)))))
+    fun recordsForSessionCreated(key: String, direction: SessionDirection): List<Record<String, SessionEvent>> {
```
Comment (nikinagy) [inline components/link-manager/src/main/kotlin/net/corda/p2p/linkmanager/sessions/events/StatefulSessionEventPublisher.kt]: Is there any reason why we retain this function here? We are not publishing anymore so having this in the publisher feels a bit off.

**Review by williamvigorr3 — DISMISSED** [review]
LGTM when @nikinagy is happy

File: components/link-manager/src/main/kotlin/net/corda/p2p/linkmanager/sessions/events/StatefulSessionEventPublisher.kt (hunk 4 above)
Comment (YashNabar) [reply components/link-manager/src/main/kotlin/net/corda/p2p/linkmanager/sessions/events/StatefulSessionEventPublisher.kt]: Addressed in [1b8b25b](https://github.com/corda/corda-runtime-os/pull/5846/commits/1b8b25bf81ea04f589e47f3841f8e059a661a91d)

**Review by nikinagy — APPROVED** [review]
LGTM

**Pull request merged**